gcli/src/cmd/config.rs

```diff
@@ -0,0 +1,158 @@
+        let conf = path.unwrap_or(Self::config()?);
+        if let Some(parent) = conf.parent() {
+            if !parent.exists() {
+                fs::create_dir_all(parent)?;
+            }
+        }
+
+        fs::write(conf, toml::to_string_pretty(self)?).map_err(Into::into)
+    }
+
+    /// NOTE: currently just a simple wrapper for [`Self::write`] since we
+    /// just have one config option.
+    pub fn exec(&self) -> Result<()> {
+        if self.action == Action::Set {
+            self.write(None)?
+        }
+
+        println!(
+            "{}: {}",
+            "RPC URL".bold(),
+            self.url.clone().unwrap_or_default().as_ref()
+        );
+        Ok(())
+    }
+}
+
+/// Config action
+#[derive(Clone, Debug, Parser, PartialEq, Eq, Default)]
+pub enum Action {
+    /// Set a config setting
+    Set,
+    /// Get current config settings
+    #[default]
+    Get,
+}
+
+/// Vara networks
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum Network {
+    /// Vara main network
+    #[default]
+    Mainnet,
+    /// Vara development network
+    Testnet,
+    /// Localhost endpoint
+    Localhost,
+    /// Customized vara network
+    Custom(Url),
+}
+
+impl AsRef<str> for Network {
+    fn as_ref(&self) -> &str {
+        match self {
+            Self::Mainnet => "wss://rpc.vara.network:443",
+            Self::Testnet => "wss://testnet.vara.network:443",
+            Self::Localhost => "ws://localhost:9944",
+            Self::Custom(url) => url.as_str(),
+        }
+    }
+}
+
+impl ToString for Network {
+    fn to_string(&self) -> String {
+        self.as_ref().into()
+    }
+}
+
+impl ValueEnum for Network {
+    fn value_variants<'a>() -> &'a [Self] {
+        &[Network::Mainnet, Network::Testnet, Network::Localhost]
+    }
+
+    fn to_possible_value(&self) -> Option<PossibleValue> {
+        match self {
+            Self::Mainnet => Some(PossibleValue::new("mainnet")),
+            Self::Testnet => Some(PossibleValue::new("testnet")),
+            Self::Localhost => Some(PossibleValue::new("localhost")),
+            Self::Custom(_) => None,
+        }
+    }
+
+    fn from_str(input: &str, ignore_case: bool) -> Result<Self, String> {
+        Ok(match input {
+            mainnet
+                if mainnet == "mainnet" || ignore_case && mainnet.to_lowercase() == "mainnet" =>
+            {
+                Self::Mainnet
+            }
+            testnet
+                if testnet == "testnet" || ignore_case && testnet.to_lowercase() == "testnet" =>
+            {
+                Self::Testnet
+            }
+            localhost
+                if localhost == "localhost"
+                    || ignore_case && localhost.to_lowercase() == "localhost" =>
+            {
+                Self::Localhost
+            }
+            _ => Self::Custom(Url::parse(input).map_err(|_| input.to_string())?),
+        })
+    }
```

Comment:
```suggestion
    fn from_str(input: &str, ignore_case: bool) -> Result<Self, String> {
        let input = if ignore_case {
            Cow::Owned(input.to_lowercase())
        } else {
            Cow::Borrowed(input)
        };

        Ok(match input.as_ref() {
            "mainnet" => Self::Mainnet,
            "testnet" => Self::Testnet,
            "localhost" => Self::Localhost,
            input => Self::Custom(Url::parse(input).map_err(|_| input.to_string())?),
        })
    }
```

Reply:
interesting, I didn't know we can just `Cow` like this before

Reply:
ddcaeb8